Runtime error: invalid memory address or nil pointer dereference

**Describe the bug**
Using promtail and latest loki a push of the data results in a runtime error: invalid memory address or nil pointer dereference (see stack trace at the bottom). After a recent new docker image this issue started, but I don't know which version I had before because I just used `latest`. I'm reasonably sure the previous `latest` was still okay.
It result in grafana showing no data at all anymore.

**To Reproduce**
Have promtail push its data to the loki push endpoint.

**Expected behavior**
The data is correctly pushed, and handles by loki.

**Environment:**
 - Infrastructure: loki on raspberry pi (arm32v7), promtail on ubuntu
 - Deployment tool: docker compose on the pi

Loki:
```
loki, version master-5d6d552 (branch: master, revision: 5d6d552d)
  build user:       root@885f2b11cf73
  build date:       2021-02-18T16:01:43Z
  go version:       go1.15.3
  platform:         linux/arm
```
Using promtail version:
```
promtail, version 2.1.0 (branch: HEAD, revision: 1b79df37)
  build user:       root@b8941a81fe55
  build date:       2020-12-24T14:08:17Z
  go version:       go1.15.3
  platform:         linux/amd64
```

**Screenshots, Promtail config, or terminal output**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11b20]

goroutine 440 [running]:
runtime/internal/atomic.goXadd64(0x3c10ee4, 0x0, 0x0, 0x80000000, 0xbff00000)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
go.uber.org/atomic.(*Int64).Sub(...)
	/src/loki/vendor/go.uber.org/atomic/int64.go:55
github.com/grafana/loki/pkg/ingester.(*replayController).Sub(0x3c10ec0, 0x0, 0x0)
	/src/loki/pkg/ingester/replay_controller.go:67 +0x44
github.com/grafana/loki/pkg/ingester.(*Ingester).removeFlushedChunks(0x3d2cdc0, 0x3e12b00, 0x3e12780, 0x488b401)
	/src/loki/pkg/ingester/flush.go:326 +0x24c
github.com/grafana/loki/pkg/ingester.(*Ingester).sweepInstance(0x3d2cdc0, 0x3e12b00, 0x100)
	/src/loki/pkg/ingester/flush.go:177 +0xd8
github.com/grafana/loki/pkg/ingester.(*Ingester).sweepUsers(0x3d2cdc0, 0x4430100)
	/src/loki/pkg/ingester/flush.go:167 +0x5c
github.com/grafana/loki/pkg/ingester.(*Ingester).loop(0x3d2cdc0)
	/src/loki/pkg/ingester/ingester.go:360 +0x94
created by github.com/grafana/loki/pkg/ingester.(*Ingester).starting
	/src/loki/pkg/ingester/ingester.go:304 +0x100
```

Just googling this error as well - new docker install with grafana/loki:latest (practically the same error):

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11b20]

goroutine 437 [running]:
runtime/internal/atomic.goXadd64(0x4838ca4, 0x0, 0x0, 0x80000000, 0xbff00000)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
go.uber.org/atomic.(*Int64).Sub(...)
	/src/loki/vendor/go.uber.org/atomic/int64.go:55
github.com/grafana/loki/pkg/ingester.(*replayController).Sub(0x4838c80, 0x0, 0x0)
	/src/loki/pkg/ingester/replay_controller.go:67 +0x44
github.com/grafana/loki/pkg/ingester.(*Ingester).removeFlushedChunks(0x4558000, 0x4711400, 0x520c200, 0x538bb01)
	/src/loki/pkg/ingester/flush.go:326 +0x24c
github.com/grafana/loki/pkg/ingester.(*Ingester).sweepInstance(0x4558000, 0x4711400, 0x100)
	/src/loki/pkg/ingester/flush.go:177 +0xd8
github.com/grafana/loki/pkg/ingester.(*Ingester).sweepUsers(0x4558000, 0x4870100)
	/src/loki/pkg/ingester/flush.go:167 +0x5c
github.com/grafana/loki/pkg/ingester.(*Ingester).loop(0x4558000)
	/src/loki/pkg/ingester/ingester.go:360 +0x94
created by github.com/grafana/loki/pkg/ingester.(*Ingester).starting
	/src/loki/pkg/ingester/ingester.go:304 +0x100

Deployed on Raspberry Pi 4 for the first time...

--- a/pkg/ingester/replay_controller.go
+++ b/pkg/ingester/replay_controller.go
@@ -40,9 +40,9 @@ type Flusher interface {
 
 // replayController handles coordinating backpressure between WAL replays and chunk flushing.
 type replayController struct {
+	currentBytes atomic.Int64
 	cfg          WALConfig
 	metrics      *ingesterMetrics
-	currentBytes atomic.Int64
 	cond         *sync.Cond
 	isFlushing   atomic.Bool
 	flusher      Flusher

--- a/pkg/ingester/replay_controller.go
+++ b/pkg/ingester/replay_controller.go
@@ -40,6 +40,12 @@ type Flusher interface {
 
 // replayController handles coordinating backpressure between WAL replays and chunk flushing.
 type replayController struct {
+	// Note, this has to be defined first to make sure it is aligned properly for 32bit ARM OS
+	// From https://golang.org/pkg/sync/atomic/#pkg-note-BUG:
+	// > On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for
+	// > 64-bit alignment of 64-bit words accessed atomically. The first word in a
+	// > variable or in an allocated struct, array, or slice can be relied upon to
+	// > be 64-bit aligned.
 	currentBytes atomic.Int64
 	cfg          WALConfig
 	metrics      *ingesterMetrics

